src/config.c

```diff
@@ -607,60 +619,268 @@ int parseCommandLine(int argc,char **argv,ParodusCfg * cfg)
-  	if (strlen(output)>0  && strcmp(output,"SUCCESS")==0)
+	CURL *curl;
+	CURLcode res;
+	CURLcode time_res;
+	struct curl_slist *list = NULL;
+	struct curl_slist *headers_list = NULL;
+
+	char *mac_header = NULL;
+	char *serial_header = NULL;
+	char *uuid_header = NULL;
+	char *transaction_uuid = NULL;
+	double total;
+
+	struct url_data data;
+	data.size = 0;
+
+	curl = curl_easy_init();
+	if(curl)
 	{
-		//Call read script 
-		execute_token_script(newToken,get_parodus_cfg()->token_read_script,len,get_parodus_cfg()->hw_mac,get_parodus_cfg()->hw_serial_number);
-	}	
-	else 
+		data.data = (char *) malloc(sizeof(char) * MAX_TOKEN_LEN);
+		if(NULL == data.data)
+		{
+			ParodusError("Failed to allocate memory.\n");
+			return -1;
+		}
+		data.data[0] = '\0';
+
+		createCurlheader(mac_header, serial_header, uuid_header, transaction_uuid, list, &headers_list);
+
+		curl_easy_setopt(curl, CURLOPT_URL, TOKEN_SERVER_URL);
+		curl_easy_setopt(curl, CURLOPT_TIMEOUT, CURL_TIMEOUT_SEC);
+		curl_easy_setopt(curl, CURLOPT_INTERFACE, get_parodus_cfg()->webpa_interface_used);
+		/* set callback for writing received data */
+		curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_callback_fn);
+		curl_easy_setopt(curl, CURLOPT_WRITEDATA, &data);
+
+		curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers_list);
+
+		/* setting curl resolve option as default mode.
+		If any failure, retry with v4 first and then v6 mode. */
+
+		if(r_count == 1)
+		{
+			ParodusInfo("Curl Ip resolve option set as V4 mode\n");
+			curl_easy_setopt(curl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
+		}
+		else if(r_count == 2)
+		{
+			ParodusInfo("Curl Ip resolve option set as V6 mode\n");
+			curl_easy_setopt(curl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V6);
+		}
+		else
+		{
+			ParodusInfo("Curl Ip resolve option set as default mode\n");
+			curl_easy_setopt(curl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_WHATEVER);
+		}
+
+		/* set the cert for client authentication */
+		curl_easy_setopt(curl, CURLOPT_SSLCERT, get_parodus_cfg()->client_cert_path);
+
+		curl_easy_setopt(curl, CURLOPT_CAINFO, get_parodus_cfg()->cert_path);
+
+		/* disconnect if it is failed to validate server's cert */
+		curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 1L);
+
+		/* Perform the request, res will get the return code */
+		res = curl_easy_perform(curl);
+		ParodusInfo("themis curl response code is %d\n", res);
+
+		time_res = curl_easy_getinfo(curl, CURLINFO_TOTAL_TIME, &total);
+		if(time_res == 0)
+		{
+			ParodusInfo("curl response Time: %.1f seconds\n", total);
+		}
+		if(transaction_uuid !=NULL)
+		{
+			free(transaction_uuid);
+			transaction_uuid = NULL;
+		}
+		if(mac_header !=NULL)
+		{
+			free(mac_header);
+			mac_header = NULL;
+		}
+		if(serial_header !=NULL)
+		{
+			free(serial_header);
+			serial_header = NULL;
+		}
+		if(uuid_header !=NULL)
+		{
+			free(uuid_header);
+			uuid_header = NULL;
+		}
+		curl_slist_free_all(headers_list);
+		if(data.data)
+		{
+			strncpy(newToken, data.data, len);
```

Comment:
This copy should be done only during HTTP status code 200

Reply:
incorporated.